docs/source/plugin/what.rst

```diff
@@ -0,0 +1,135 @@
+=================
+What is a Plugin?
+=================
+
+Before writing your first plugin, you may want to know what is a plugin, how is
+it made, and what are the best solutions for your need. In this chapter we'll
+cover what is a plugin, what you can do with them, and some vocabulary.
+
+You may want to skip ahead to :doc:`anatomy` if you already know what a plugin
+is and what types of plugin there are.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Plugin Types
+============
+
+There are four types of plugins: :term:`Single file plugin`,
+:term:`Folder plugin`, :term:`Namespace package plugin`, and
+:term:`Entry point plugin`.
+
+Single file
+-----------
+
+A **Single file** plugin is the most basic form a Sopel plugin can take. It is
+composed of a single file available to Sopel from its plugins directory (or
+from one of its
+:attr:`extra directories <sopel.config.core_section.CoreSection.extra>`). You
+should write a single file plugin when:
+
+* everything is in a single file
+* you don't have any dependencies other than Sopel and the Python's built-in
+  library
+
+This type of plugin is easy to install and share: copy its file into the right
+location, and Sopel will load it on startup.
+
+Folder
+------
+
+A **Folder** plugin is a folder that contains a ``__init__.py`` file. It is
+available to Sopel the same way a :ref:`Single file` plugin is, and shares the
+same limitation: Everything that is not available in this ``__init__.py`` file
+won't be visible to Sopel.
+
+By default, Sopel's plugin directories are **not** available in ``sys.path``,
+so it is not possible to do ``from my_plugin_dir import some_module`` without
+a specific configuration from the bot's owner.
+
+When a plugin author wants to split code into different files, and/or share it
+between multiple plugins, the best option is to use an :ref:`Entry point`
+plugin instead.
+
+Namespace package
+-----------------
+
+A **Namespace package** plugin is a Python :term:`namespace package`, using
+the namespace ``sopel_modules``. It must be installed in the Python environment
+to be used by Sopel, and can require Python dependencies.
+
+Given a ``sopel_modules.plugin`` plugin, Sopel will load everything that is
+available from the ``sopel_modules/plugin/__init__.py`` file.
+
+It is the initial version of Sopel's packaged plugins: it can be packaged and
+uploaded to `PyPI`_ then installed using ``pip install``.
+
+When a plugin author wants to distribute a Sopel plugin, the best option is to
+use an :ref:`Entry point` plugin instead.
+
+Entry point
+-----------
+
+.. versionadded:: 7.0.0
```

Comment:
```suggestion
.. versionadded:: 7.0
```

I can't think of any places we include a `.z` version in these annotations; it's (pretty much?) always `x.y` only.

Reply:
Oh yeah, why did I do that?